### ISSUE TITLE
NodeJS built-in package import throw an error with this plugin

### ISSUE THREAD

**Issue by pebie** [microsoft/vscode-eslint#1633]
**Eslint configuration :**

```yaml
root: true
parserOptions:
  ecmaVersion: 13
  sourceType: module
env:
  node: true
  es6: true
  mocha: true
plugins:
  - 'import'
extends:
  - 'eslint:recommended'
  - 'plugin:prettier/recommended'
  - 'plugin:import/errors'
  - 'plugin:import/warnings'
  - 'plugin:security/recommended'
settings:
  'import/internal-regex': '^\.\.?/'
rules:
  'no-console': 'off'
  'no-duplicate-imports': 'error'
  'no-return-await': 'error'
  'no-unneeded-ternary': 'error'
  'no-unused-vars': ['error', {ignoreRestSiblings: true, args: 'none'}]
  'prefer-object-spread': 'error'
  'require-await': 'off'
  'import/order':
    [
      'error',
      {
        groups: ['builtin', 'external', ['internal', 'parent', 'sibling', 'index'], 'unknown'],
        'newlines-between': 'always',
        alphabetize: {order: 'asc', caseInsensitive: true},
      },
    ]
```

**My ESM module looks like this** 

```javascript
import test from 'node:test'
// Whatever...
```

**Run CLI eslint command :** 
yarn
```shell
yarn eslint ./
```
This command run eslint against all of my *.js files and does not cause an error

**With vscode-eslint plugin**
```shell
Unable to resolve path to module 'node:test'.
```

In the plugin configuration, I'm trying to figure out what's wrong. I let default setup. 

**Node version :** 18.16
**Eslint informations :** 
```json
        "eslint": "8.31.0",
        "eslint-config-prettier": "8.6.0",
        "eslint-plugin-import": "2.27.5",
        "eslint-plugin-prettier": "4.2.1",
        "eslint-plugin-security": "1.5.0",
```

**Additionnal info about package.json** : I use `"type": "module"`

Thanks for your help !

**Comment by dbaeumer** [comment]
The `node:test` module is only available in node 19.x to my knowledge. So you need to use a custom node version (not the one that ships with Electron) by using the `eslint.runtime` setting.

**Comment by robin-thomas** [comment]
> The node:test module is only available in node 19.x to my knowledge.

@dbaeumer Thats incorrect. Its available in `18.x` https://nodejs.dev/en/api/v18/test/ and `16.x` as well

**Comment by dbaeumer** [comment]
You are right. However in all version it is flagged as experimental and it needs an additional parameter to work `--test`. Run can pass additional parameters to Node using the `eslint.execArgv` setting.

**Comment by robin-thomas** [comment]
@dbaeumer Yes, it's marked as experimental in `18.x` and `19.x`, but it's already stable in `20.x`.

For now, I have added the below in my `.eslintrc.yml` to ignore the error:
```
settings:
  import/core-modules: [ node:test ]
```

**Comment by dbaeumer** [comment]
The thing is that VS Code doesn't ship with NodeJS version 20.x.
